client/connectivity/connectivity.go

```diff
@@ -221,36 +221,33 @@ func (c *circle) newPeer(ctx context.Context, p *pb.Peer) *Peer {
 }
 
 func (c *circle) dialPeer(ctx context.Context, addr string) (net.Conn, error) {
-	if strings.HasPrefix(addr, "udp:") {
-		return c.udpSocket.DialContext(ctx, strings.TrimPrefix(addr, "udp:"))
-	} else {
+	sp := strings.SplitN(addr, ":", 2)
+	addr = sp[1]
+	switch pb.Endpoint_Type(pb.Endpoint_Type_value[sp[0]]) {
+	case pb.Endpoint_SCTP_OVER_UDP:
+		return c.udpSocket.DialContext(ctx, addr)
+	case pb.Endpoint_TCP:
 		var d net.Dialer
 		return d.DialContext(ctx, "tcp", addr)
+	default:
+		return nil, fmt.Errorf("internal error: unknown endpoint type: %v", sp[0])
```

Comment:
By adding a log statement here (and commenting out the TCP endpoint case above, essentially making it unsupported) we confirmed that gRPC implements a back-off on failing dials. So, this error here will not result in a tight-loop.